modules/xfem/src/auxkernels/LineSegmentLevelSetAux.C

```diff
@@ -0,0 +1,108 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LineSegmentLevelSetAux.h"
+#include "LineSegmentCutSetUserObject.h"
+#include "FEProblem.h"
+
+registerMooseObject("MooseApp", LineSegmentLevelSetAux);
+
+template <>
+InputParameters
+validParams<LineSegmentLevelSetAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addClassDescription(
+      "Auxiliary Kernel that calcuates level set value using line segments' description.");
+  params.addParam<UserObjectName>(
```

Comment:
Since it's a `LineSegmentCutSetUserObject`, something like `line_segment_cut_set_user_object` would be more appropriate for the parameter name.